src/krylov_phiv_chainrules.jl

```diff
@@ -0,0 +1,42 @@
+function ChainRulesCore.frule((_, Δt, ΔA, Δb), ::typeof(expv), t, A, b; kwargs...)
+    w = expv(t, A, b; kwargs...)
+    ∂w = similar(w)
+    mul!(∂w, A, w)
+    ∂w .*= Δt
+    if !isa(Δb, AbstractZero)
+        ∂w .+= expv(t, A, Δb; kwargs...)
+    end
+    # TODO: handle ΔA
+    ΔA isa AbstractZero || error("ΔA currently cannot be pushed forward")
+    return w, ∂w
+end
+
+function ChainRulesCore.rrule(::typeof(expv), t, A, b; kwargs...)
+    w = expv(t, A, b; kwargs...)
+    function expv_pullback(Δw)
+        ∂t = Thunk() do
+            t̄ = A isa AbstractMatrix ? conj(dot(Δw, A, w)) : dot(mul!(similar(w), A, w), Δw)
+            return t isa Real ? real(t̄) : t̄
+        end
+        # TODO: handle ∂A
+        ∂A = @thunk error("Adjoint wrt A not yet implemented")
+        ∂b = Thunk() do
+            # using similar is necessary to ensure type-stability
+            b̄ = similar(b)
+            _copyto!(b̄, expv(t', A', Δw; kwargs...))
```

Comment:
`adjoint` is currently not part of the required interface for generic `A`. How should we handle this?
- Make it a required part of the interface
- Make it an optional requirement for `A` (i.,e. "if you want to work with Zygote, you also need to define `adjoint`")
- restrict this `rrule` to `A::AbstractMatrix` (this may be necessary anyways, as for non-matrix `A`, its differential type is unknown.).

Reply:
>"if you want to work with Zygote, you also need to define adjoint"

I think that's entirely reasonable.